docs/tutorials/adding-a-module-to-your-runtime.md

```diff
@@ -0,0 +1,306 @@
+
+To see how these features actually get used in the runtime code, we can open `contract-chain/runtime/src/lib.rs`:
+
+**lib.rs**
+```rust
+//! The Substrate Node Template runtime. This can be compiled with `#[no_std]`, ready for Wasm.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(not(feature = "std"), feature(alloc))]
+// `construct_runtime!` does a lot of recursion and requires us to increase the limit to 256.
+#![recursion_limit="256"]
+
+#[cfg(feature = "std")]
+use serde::{Serialize, Deserialize};
+use parity_codec::{Encode, Decode};
+...
+```
+
+You can see that at the top of the file, we define that we will use `no_std` when we are *not* using the `std` feature. A few lines lower you can see `#[cfg(feature = "std")]` above the `serde` import, which is a flag saying to only use `serde` when we have enabled the `std` feature. We need to do this because `serde` has dependencies on the Rust standard libraries, and thus cannot be used in `no_std` environments.
+
+> Note: The config attribute only applies to the import directly below it, so in the snippet above, `parity_codec` is always used in both the `std` and `no_std` environments.
+
+### Importing the Contract Module Crate
+
+Okay, now that we have covered the basics of crate features, we can actually import the Contract module. The Contract module is probably the most complicated module in the SRML, so it makes for a good example of some of the trickiness that can be involved when adding additional modules. To give you a hint as to what is to come, you should take a look at the [`Cargo.toml` file for the Contract module](https://github.com/paritytech/substrate/blob/v1.0/srml/contract/Cargo.toml).
+
+First we will add the new dependency by simply copying an existing module, and changing the values. So based on the `balances` import shown above, my `contract import will look like:
+
+```rust
+[dependencies.contract]
+default_features = false
+git = 'https://github.com/paritytech/substrate.git'
+package = 'srml-contract'
+rev = '783ca1892892454e05e234cda5f7a2e42a54461e'
+```
+
+You can see that the Contract module has an `std` feature, thus we need to add the `contract/std` feature to the `std` feature of our runtime:
+
+```rust
+std = [
+    ...
+    'contract/std',
+]
+```
+
+If you forget to set the `contract/std` feature, when building to your native binaries you will get errors like:
+
+```rust
+error: cannot find macro `vec!` in this scope
+   --> /Users/shawntabrizi/.cargo/git/checkouts/substrate-7e08433d4c370a21/783ca18/core/sr-sandbox/src/../without_std.rs:290:24
+    |
+290 |         let mut return_val = vec![0u8; sandbox_primitives::ReturnValue::ENCODED_MAX_SIZE];
+    |                              ^^^
+
+error: aborting due to previous error
+```
+
+If we look closer, it also has another feature: `core`.
```

Comment:
I see this on master, but not on v1.0 which is what this tutorial is targeting, right?

Reply:
v1.0 compiles fine with or without the core feature.